src/store/CrosshairStore.js

```diff
@@ -102,7 +91,7 @@ class CrosshairStore {
         const bar = stx.barFromPixel(stx.cx);
         const data = stx.chart.dataSegment[bar];
         if (!data) {
-            this.hide();
+            this.updateTooltipPosition({ left: -5000, top: 0, rows: null });
```

Comment:
may i know the reason why are we doing this? instead of hide?

Reply:
@cakasuma I'm going to make a wild guess and say because it performs better.

Reply:
> @cakasuma I'm going to make a wild guess and say because it performs better.

If that's the case, why is it faster?

Reply:
@brandon-binary , refer to comments made for the implementation change [here](https://github.com/binary-com/SmartCharts/pull/816/files#diff-b630a048c5c25a74efd1deaf1c0fbb01R7) and [here](https://github.com/binary-com/SmartCharts/pull/816/files#diff-2f67da90737005346bff0e111831096cR355).

Reply:
@brandon-binary the `hide` function was setting top and left, which forces the browser to trigger a layout recalculation.

The `transform: ` property doesn’t alter the page layout, it’s a post layout property. 

That’s why it’s faster